src/NitroSharp/Content/ArchiveManager.cs

```diff
@@ -0,0 +1,203 @@
+                foreach (KeyValuePair<string, VFSNode> pair in Children)
+                {
+                    pair.Value.Dispose();
+                }
+            }
+
+            if (MountedArchives != null)
+            {
+                foreach (IArchiveFile mountedArchive in MountedArchives)
+                {
+                    mountedArchive.Dispose();
+                }
+            }
+        }
+    }
+
+    internal sealed class ArchiveException : Exception
+    {
+        public ArchiveException()
+            : base("Unable to open the archive")
+        {
+        }
+
+        public ArchiveException(string format, string message)
+            : base($"{format} : {message}")
+        {
+        }
+
+    }
+
+    internal sealed class ArchiveManager : IDisposable
+    {
+        public static readonly Encoding DefaultEncoding;
+
+        private readonly VFSNode _root;
+        private readonly Encoding _encoding;
+
+        static ArchiveManager()
+        {
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+            DefaultEncoding = Encoding.GetEncoding("shift-jis");
+        }
+
+        public ArchiveManager(Encoding? encoding = null)
+        {
+            if (encoding == null)
+            {
+                encoding = DefaultEncoding;
+            }
+            _encoding = encoding;
+            _root = new VFSNode();
+        }
+
+        public void Dispose()
+        {
+            _root.Dispose();
+        }
+
+        public void AddMounts(MountPoint[] mountPoints, string rootDirectory, ContentManager content)
+        {
+            foreach (MountPoint mountPoint in mountPoints)
+            {
+                AddMount(mountPoint, rootDirectory, content);
+            }
+        }
+
+        private void AddMount(MountPoint mountPoint, string rootDirectory, ContentManager content)
+        {
+            string[] splitedPath = mountPoint.MountName.ToLowerInvariant().Split("/");
+
+            VFSNode head = _root;
+            foreach (string part in splitedPath)
+            {
+                if (head.Children == null)
+                {
+                    head.Children = new Dictionary<string, VFSNode>();
+                }
+
+                if (!head.Children.ContainsKey(part))
+                {
+                    head.Children[part] = new VFSNode();
+                }
+
+                head = head.Children[part];
+            }
+
+            if (head.MountedArchives == null)
+            {
+                head.MountedArchives = new List<IArchiveFile>();
+            }
+
+            string fullPath = Path.Combine(rootDirectory, mountPoint.ArchiveName);
+            if (File.Exists(fullPath))
+            {
+                Func<MemoryMappedFile,Encoding,IArchiveFile?>[] loadMethods = new Func<MemoryMappedFile,Encoding,IArchiveFile?>[]
+                {
+                    (m, e) => NPAFile.TryLoad(m, e),
+                    (m, e) => AFSFile.TryLoad(m, e),
```

Comment:
What exactly was the point of changing ``NPAFile.TryLoad`` to ``(m, e) => NPAFile.TryLoad(m, e)``? I'm not necessarily asking you to revert this change (I can do that myself), just wondering.